receiver/tlscheckreceiver/config.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package tlscheckreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tlscheckreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tlscheckreceiver/internal/metadata"
+)
+
+// Predefined error responses for configuration validation failures
+var (
+	errMissingUrl = errors.New(`"url" must be specified`)
+	errInvalidUrl = errors.New(`"url" must be in the form of <scheme>://<hostname>[:<port>]`)
+)
+
+// Config defines the configuration for the various elements of the receiver agent.
+type Config struct {
+	scraperhelper.ControllerConfig `mapstructure:",squash"`
+	metadata.MetricsBuilderConfig  `mapstructure:",squash"`
+	Targets                        []*targetConfig `mapstructure:"targets"`
+}
+
+type targetConfig struct {
+	Url string `mapstructure:"url"`
+}
+
+// Validate validates the configuration by checking for missing or invalid fields
+func (cfg *targetConfig) Validate() error {
+	var err error
+
+	if cfg.Url == "" {
+		err = multierr.Append(err, errMissingUrl)
+	} else {
+		_, parseErr := url.ParseRequestURI(cfg.Url)
+		if parseErr != nil {
+			err = multierr.Append(err, fmt.Errorf("%s: %w", errInvalidUrl.Error(), parseErr))
+		}
+	}
+
+	return err
+}
+
+// Validate validates the configuration by checking for missing or invalid fields
+func (cfg *Config) Validate() error {
+	var err error
+
+	if len(cfg.Targets) == 0 {
+		err = multierr.Append(err, errMissingUrl)
+	}
+
+	for _, target := range cfg.Targets {
```

Comment:
From memory, the `component.ValidateConfig(...)` method performs reflection instead of type inference so it can explore each field if they are set. 

It might be possible that your Targets are being evaluated twice, do you mind just double checking this isn't the case?

Reply:
Looking at the test cases, it looks like it doesn't evaluate slices so I think it is fine to ignore this comment :)